### ISSUE TITLE
Cookie banner: FSE friendly way to enable and modify the banner

### ISSUE THREAD

**Issue by simison** [Automattic/wp-calypso#52100]
Paid sites get a feature called "cookie banner". The code is Jetpack plugin and you enable & modify it via Customizer's widgets ([docs](https://wordpress.com/support/cookie-widget/)).

<img width="1481" alt="Screenshot 2021-04-20 at 13 33 47" src="https://user-images.githubusercontent.com/87168/115382634-94ba1700-a1dd-11eb-9aca-30b50038fe0f.png">

Widget is a bit odd place to control a site-wide banner like this because you place it at specific widget area but the banner appears on top of the site. Customizer is great though because it offers a live preview.

FSE sites don't have Customizer nor widgets, and don't get the feature.

### Ideas

Some could be implemented across both Classic and FSE themes.

- **Settings**
    - Wouldn't necessarily include a site-wide preview. We could still have a simple inline preview of course.
    - Example from Wix:

        <img width="600" alt="Screenshot 2022-12-06 at 15 04 04" src="https://user-images.githubusercontent.com/87168/205920246-758d8600-83d4-4ba3-9eab-138429ef3206.png">

- **Privacy tab**
    - Like above but instead of Settings, show a tab at core's "Settings → Privacy" page (currently Atomic only) as a tab:
 
    <img width="600" alt="Screenshot 2022-12-06 at 14 53 19" src="https://user-images.githubusercontent.com/87168/205918303-3616bcea-4fd6-4e48-b19b-258e8e9ba4aa.png">

- **In site editor as a panel**
   - Preferably some core-first UI for manipulating floating elements (such as banners, modals). Does not exist yet.
   - New section at Template settings, Plugins icon row, or new tab under Settings:

      <img width="300" alt="Screenshot 2022-12-06 at 14 58 06" src="https://user-images.githubusercontent.com/87168/205919258-ab3b9ac1-ca1d-4494-b1c0-303e7a4c1ced.png">
   - Could allow more direct wysiwyg, block-style manipulation instead of fiddling with settings form.
 
- **In site editor as a block**
   - Similar to widgets feels odd to place block that then hovers on top of the page, but this is quite pragmatic solution. There's [some mechanisms comign up in core to support "fixed" positioning of blocks](https://github.com/WordPress/gutenberg/discussions/46032#discussioncomment-4343350).

- **Customizer section**
    - Add new link to Customizer from sidebar and call it "privacy banner" or similar — re-use Widget's settings but without Widget mechanisms, just directly in the Customizer sidebar.
    - Not so great to rely on customizer when we have FSE.
 
Related feature request https://github.com/Automattic/wp-calypso/issues/64768

**Comment by arcangelini** [comment]
I think it may be best to get someone from ads or legal to weigh in on the need for this widget. I believe the features of this widget are extremely basic and do not meet the needs of GDPR any longer. The cookies that it sets when you "accept" the banner were only for advertising purposes — and we no longer read them because we have an updated GDPR banner that actually does give users some choice into the use of their data. We currently only show this specific banner for US-based visitors or sites that add the widget manually and aren't part of WordAds. In those cases, it's more of a cookie notice and doesn't actually have any useful functionality.

WordAds now has a specially designed cookie banner (pMyNb-4TC-p2) that is displayed apart from this widget. I believe @catscheer (also tagging @jennyzhu in her absence) just made a recent update as well regarding the state of the banners (p2y3YZ-4RJ-p2). With all this in mind, it may be time to deprecate/let this go.

Tagging @ivan-ottinger who is working on the widgets currently.

**Comment by ivan-ottinger** [comment]
If we decide to deprecate the _Cookie & Consents Banner_ widget, we will be glad to help with that in @Automattic/apex. 👍🏼

**Comment by simison** [comment]
Convo p4H3ND-18w-p2 leaning towards keeping this de-facto deprecated going forward, even if it works on widget areas.

**Comment by ivan-ottinger** [comment]
> Convo p4H3ND-18w-p2 leaning towards keeping this de-facto deprecated going forward, even if it works on widget areas.

Just to double-check I understand correctly - the decision hasn't been fully made yet, right? Thanks.

**Comment by simison** [comment]
Still open about it but it definitely lowered the priority significantly in my mind.

Sounds like it would need entirely be redesigned to actually be useful, and in it current form it isn't helpful, other than on the surface. I'm all ears for more input on this tho! Especially if customers are asking for something like this, and what are they specifically asking/needing.

**Comment by filipanoscampos** [comment]
Here is a case of a user who needs to add a Cookie banner and is using a theme without a Widget area: 4777334-zd-woothemes

I am sharing this here since the previous issues were closed in favor of this one. 
It would be great to have an alternative to offer our users, and not leave them without a solution.

**Comment by philnick206** [comment]
Adding this issue as another example of a customer wanting this: #4800058-zen

The use-case here is to have compliance with Google Analytics which is integrated on a simple site. I assume that the current widget wouldn't be compliant enough for GA, thus this seems like a legit scenario that is missing coverage, but I'm out of my depth.

**Comment by rw-ye** [comment]
Another case here where the user is asking about adding a cookie banner here: 4817167-zd-woothemes

They are using FSE

**Comment by MaxPhilip** [comment]
Another user wanting to add a cookie banner using FSE with not widget area: 4811882-zen

**Comment by daria2303** [comment]
4821868-zd-woothemes

**Comment by upwardmomentum84** [comment]
Another user wanting to add a cookie banner using FSE - 4828543-zd-woothemes

**Comment by JessBoctor** [comment]
I had another user wanting to use this widget in 32880209-hc. They ran into the problem of the Russell theme not having a widget area. 

Seeing this discussion, I asked the user for feedback of why they wanted to use this widget. Their reply was, "I need to inform visitors that the site uses cookies, as I am adding Google Analytics code"

**Comment by dolgelukkig** [comment]
Another one here: 34442779-hc, the theme they are using is Attar

**Comment by zdenys** [comment]
4851074-zd-woothemes

**Comment by csabarakasz** [comment]
4829352-zd-woothemes

User is on Premium plan, using Payton theme and would need the Cookies banner. Suggested switching themes and added their case here!

**Comment by philnick206** [comment]
It feels like we should no longer be recommending ANY sites use the Jetpack Cookies and Consent Widget (since it is no longer compliant with GDPR and related services and is no longer used by WordAds). 

If that is the case, then there is a gap for any simple sites that are not in WordAds regardless of FSE or not. The only question I guess would be whether the current WP.com CMP that is automatically enabled on WordAds sites also fulfills the requirements for any services like Google Analytics (or similar, like Cloudflare Analytics, are there others)? 

And there is a gap on WoA but we can recommend they use a consent management plugin (I'm not sure if we have a shortlist of recommended options or not).

Did I get all that right @ivan-ottinger @simison @arcangelini ?

**Comment by ivan-ottinger** [comment]
Hey @philnick206 👋🏼  

> Did I get all that right?

While I don't feel to be the person qualified to answer your question, it sounds right to me.

There were a couple of comments on the P2 post that share a bit more details: p4H3ND-18w-p2#comment-2540 (if you haven't read them already).

**Comment by gabriel-fuentes** [comment]
4867639 zd

**Comment by CGA8C** [comment]
34788451 HC

User asked for a Cookie Consent Widget

**Comment by upwardmomentum84** [comment]
6510482 HC

User asked for a Cookie Consent Widget

**Comment by donalirl** [comment]
#52100-hc concerned about not being able to add a cookie banner. I promised we'd let them know when a new solution is ready.

**Comment by csabarakasz** [comment]
4914228-zd-woothemes

**Comment by filipanoscampos** [comment]
Not an FSE friendly cookie widget, but a request for a Cookie banner that compiles with Italian law:
4882757-zd-woothemes

**Comment by kspilarski** [comment]
4915537-zd 

FSE theme: Twenty Twenty Two.

**Comment by csabarakasz** [comment]
4917434-zd-woothemes

Premium plan, no way to add cookies banner.

**Comment by jamiepalatnik** [comment]
Another report: 31226778-hc

Site is using Arbutus theme. The user mentioned "I think I am obliged to do it because of GDPR"

No alternative to offer other than switching to another theme.

**Comment by ezebecke** [comment]
Another report: 4929715-zd

Premium plan with "Byrne" theme.

**Comment by yashitamittal11** [comment]
Another report - #4935143-zen
Premium plan with FSE theme Russell

**Comment by syhussaini** [comment]
Here is another case where the user asks for Cookies and a Consent Banner #4940939-zen.

I guess we can build something into all the FSE based themes that should allow the users to activate/deactivate the cookies and consent banner with a toggle button in theme settings. 

_I am just thinking out loud_

**Comment by formosattic** [comment]
The resolution seems to require more discussion involving different parties, see internal thread: pc4f5j-1F8-p2
I've also flagged this for the DotCom janitors at pc4f5j-1F8-p2#comment-2293

**Comment by masperber** [comment]
I want to flag that our [support document on the Cookies and Consent Widget](https://wordpress.com/support/widgets/cookie-widget/) says the following:

> Themes that support the [Site Editor](https://wordpress.com/support/site-editor/) do not use widgets, so you won’t see Appearance → Widgets in your [dashboard](https://wordpress.com/home). Instead, you can edit all of your site in one place at Appearance → Editor. [Click here to learn how to use the Site Editor.](https://wordpress.com/support/site-editor/)

This seems to imply that there is a way to add the Cookies and Consent Widget in the Site Editor, but that doesn't seem to be possible currently.

**Comment by csabarakasz** [comment]
4958120-zd-woothemes Site with free plan and Remote FSE theme.

**Comment by paulostp** [comment]
4960117-zd

Personal plan with Quadrat White

**Comment by i11za** [comment]
Another report here: 35219220-hc

Site with eCommerce plan / Twenty Twenty Two theme. Recommended to use a plugin instead.

**Comment by jamiepalatnik** [comment]
Another report: [31226778-hc](35213338-hc)

Site is using Russell theme.

No alternative to offer other than switching to another theme.

**Comment by adelineyaw** [comment]
Another report: 5031409-zen
Site is using Blank Canvas (Personal plan)

**Comment by devNigel** [comment]
Another case: 5308833-zd-woothemes

The site is using FSE theme Dora on the Starter plan. Informed they have to upgrade to Pro plan and use a plugin.

**Comment by sarahcada** [comment]
Another case at 34702400-hc 
Site has no paid plan at the moment, was using Dorna theme. They may change themes.

**Comment by catscheer** [comment]
@simison and @ivan-ottinger weighing in on this from a legal perspective. If you haven't read the answer from Holly to whether or not we still need the C&C banner, you can find it here https://wp.me/p4H3ND-18w. The TL;DR was that yes, we still need to provide some kind of cookie banner option for users to enable on sites that aren't showing our ads. The point of the banner is not to control any cookies, but rather to provide the opportunity for site owner's to provide a kind of cookie notice to their site visitors informing them of the fact that there are cookies in use on the site that can't be turned off. As is clear from this thread, lots and lots of users are demanding this functionality.

**Comment by arunsathiya** [comment]
- [ ] 5380722-zen

**Comment by yansern** [comment]
> TL;DR was that yes, we still need to provide some kind of cookie banner option for users to enable on sites that aren't showing our ads. 

I feel that the "when & where to show" the cookie banner is the difficult part to implement.

If we take this difficult part (when & where to show) out of the equation, perhaps we can start by first creating (or migrating the legacy widget) to a Cookie Banner Block and let users decide by themselves if they want to it to their site (either to FSE or Widgets Editor).

At least when the functionality is there, we can come back and revisit the "when & where to show" question.

**Comment by mxhassani** [comment]
One more request here: 36152776-hc
WoA, suggested using a plugin

**Comment by jorpdesigns** [comment]
- [ ] 5462966-zen

**Comment by sophiegyo** [comment]
This came up in 36849726-hc as well.
Editing: Added Business plan for three months so they could install Cookie Banner plugin.

**Comment by liviopv** [comment]
- [ ] 5564296-zen

Simple site, Russell.

@simison @blackjackkent @yansern has there been any progress on this? I'm bumping the priority on this, considering it affects a whole continent of possible customers and basically renders the design selection in onboarding useless for Premium Plan customers and below.

**Comment by syhussaini** [comment]
Another instance here: 5577835-zd

**Comment by jorpdesigns** [comment]
Another case here: 33979137-hc

**Comment by radtechgh** [comment]
Another here: 37150572-hc

**Comment by thekingsprojects** [comment]
Another request here: 5606142-zen

**Comment by philnick206** [comment]
Another request 5610388-zen

**Comment by marianearchimbaud** [comment]
The issue was reported here as well: 37362950-hc
I created follow-up here: 5661299-zd

**Comment by geekinthegirl** [comment]
One more request here: 37650252-hc

**Comment by wpshellbelle** [comment]
One more request here: 37800885-hc

**Comment by wpshellbelle** [comment]
Another request at 37656670-hc

**Comment by dragstor** [comment]
> This came up in 36849726-hc as well. Editing: Added Business plan for three months so they could install Cookie Banner plugin.

Extended the user's Business plan for 3 more months, after discussing with @rachelwinspear :)

37679562-hc

**Comment by obenland** [comment]
@daledupreez Is there any update on the progress on this? Is Serenity still the best place for this ticket?

**Comment by jmdodd** [comment]
Another request in 5747871-zen.

**Comment by vinnykaur** [comment]
Another one in 5756817-zen

**Comment by thesacredpath** [comment]
Another request in 5754909-zen

**Comment by thesacredpath** [comment]
@obenland To my way of thinking, the cookies and consent should be on for all sites by default since you never know where visitors are going to come from. I'm wondering if it may be best to move this out of the realm of widgets and blocks, and perhaps add it directly within calypso interface, perhaps under settings. Just a thought.

**Comment by carolframen** [comment]
Another request 5764659-zd

**Comment by simison** [comment]
> should be on for all sites by default since you never know where visitors are going to come from. 

Not every site has cookies, though. Most sites shouldn't have any cookies. If they do (because of ads or such) we already add a suitable cookie banner from .com side automatically.

**Comment by ariel-maidana** [comment]
If a user enables Google Analytics on a simple site will a banner be automatically added?

**Comment by simison** [comment]
>If a user enables Google Analytics on a simple site will a banner be automatically added?

Doing a quick test; seems not.

**Comment by vajrasar** [comment]
Another 5785798-zen

**Comment by burgerberg** [comment]
Another 5810289-zen

**Comment by carolframen** [comment]
Another one 5828521-zen

**Comment by liviopv** [comment]
- [ ] 5884480-zd-woothemes

**Comment by caimin-j** [comment]
@simison @blackjackkent @yansern @daledupreez 

Could we please have an answer to @liviopv's question from [September last year](https://github.com/Automattic/wp-calypso/issues/52100#issuecomment-1250255913)?

This is an issue, open since April 2021, that affects a large number of users across an entire continent. There are also potentially legal issues since it's connected with GDPR.

If we're bugging the wrong people, please let us know who we should bug instead 😉 

Thank you!

**Comment by burgerberg** [comment]
Another user: 5906845-zen

**Comment by arinoch** [comment]
Giving this a bump with this user. 5904177-zd-woothemes

Accounts and Privacy perspective: Like the google fonts issue (https://github.com/Automattic/wp-calypso/issues/69395), the cookies and consents banner for FSE sites has the potential to be a significant driver of tickets and chats - more so than we've seen since April 2021. We waited too long to react to the google fonts changes - I would like to see us not make that mistake with this issue.

**Comment by danieldanilov** [comment]
Another customer request: 5915701-zen

**Comment by bryandev602-gh** [comment]
Another customer request: 5949366-zen

**Comment by cat-og** [comment]
Another request: 39740855-hc

**Comment by mzitinfo** [comment]
Another on in 5974060-zen 

(this is a fairly common issue in the German queue)

**Comment by carolframen** [comment]
Another one #5951263-zen

**Comment by jorgefilipecosta** [comment]
Hi, I just read this discussion and I'm sharing some insights in case they may be helpful. 
I think long term we should develop something native that works with FSE, it could even be in a plugin like a jetpack or a separate plugin allowing the wider .org community to take advantage of it.
A possible simple solution is a server-side render block that just calls the same widget code. Obviously, it would be better to have a panel in the site editor to enable disabling this as @simison referred, or a settings page because a block requires that we change every template file where we want the banner to appear which is not ideal.

In terms of a possible immediate solution right for the people needing this feature. Core has a block that allows us to use a widget as if it was a block the "Legacy Widget" block. That block is not exposed by default but @justintadlock created a plugin that exposes it on the site editor https://wordpress.org/plugins/x3p0-legacy-widget/ so I guess we could try using this plugin plus the "Cookies & Consent Banner" widget. If we don't want to install plugins it is also possible to use legacy widgets by going to the code editor and appending its code directly:
```
<!-- wp:legacy-widget /-->
```
It Should add a legacy widget block and when the user changes back to visual view it should be possible to pick the "Cookies & Consent Banner" widget (and or all the other widgets present on the website).

**Comment by mxhassani** [comment]
Great TIL @jorgefilipecosta ! 
Coincidently we now have this new update: p7DVsv-gGa-p2 shipping soon.

**Comment by bryandev602-gh** [comment]
Another one: 6017418-zen.

**Comment by alshakero** [comment]
Closing as shipped. 

See p7DVsv-gGa-p2 for context.
